eclipse/src/saros/editor/EditorActivityDelayer.java

```diff
@@ -0,0 +1,89 @@
+package saros.editor;
+
+import java.util.Objects;
+import org.apache.log4j.Logger;
+import org.eclipse.swt.widgets.Display;
+import saros.activities.EditorActivity;
+import saros.activities.TextEditActivity;
+import saros.activities.TextSelectionActivity;
+import saros.activities.ViewportActivity;
+import saros.session.AbstractActivityProducer;
+import saros.session.ISarosSession;
+import saros.ui.util.SWTUtils;
+import saros.util.StackTrace;
+
+/**
+ * This class is responsible for delaying and discarding certain activities that are normally
+ * generated frequently.
+ */
+// TODO Viewport activities are also fire to frequently, worst case is selecting text with mouse
+// while scrolling up or down
+
+final class EditorActivityDelayer extends AbstractActivityProducer {
+
+  private static final int DELAY = 100; // ms
+
+  private static final Logger log = Logger.getLogger(EditorActivityDelayer.class);
+
+  private final Display display;
+
+  private TextSelectionActivity currentSelectionActivity = null;
```

Comment:
Unnecessary? As far as I know, every instance variable is automatically initiallized with `null`.

Also, am I correct in the assumption that the SWT access restriction used to prevent concurrent access to this variable? (Otherwise, this could lead to issues with the delayed execution of `flushCurrentSelectionActivity()`.)

Reply:
See other comment, thread access in Eclipse is always a good guess.